src/main/scala/io/iohk/praos/crypto/package.scala

```diff
@@ -59,5 +59,44 @@ package object crypto {
   /**
     * The seed for a PRNG.
     */
-  type Seed = Int
+  type Seed = RandomValue
+
+  /**
+    * Operation used to "combine" seeds. According to the Praos Formalization, it should be associative.
+    * @note Currently implemented as the number obtained by concatenating the binary representation of both arguments.
+    */
+  def combineSeeds(x: Seed, y: Seed): Seed = {
+    Integer.parseInt(Integer.toBinaryString(x) ++ Integer.toBinaryString(y), 2)
+  }
+
+  /**
+    * A calculator of hash values.
+    */
+  trait Hasher {
```

Comment:
I'm not sure if this trait is needed, are we expecting to extend it later?

Reply:
If we want to allow different implementations of the hash function to be used without code changes, then I think it's a nice-to-have. Plus, this allows dependency injection.

Reply:
Maybe I missunderstood something but doesn't the `PredefinedHasher` allow changing the hash function used?

Reply:
Yes, it does, but to some extent. `PredefinedHasher` is parametrized by the name of a **predefined** algorithm (i.e. provided by `java.security.MessageDigest` such as MD5, SHA-1, etc.). If we want to write our own specific hasher (e.g. a mock hasher), then `Hasher` makes sense.

Reply:
Are we expecting to write our own specific hasher in the near future? I agree that having a mock hasher could be benefitial, however I would prefer sticking to the simpler alternative if there's no need right now to make our code more complex.

Reply:
Either way, maybe we could also move this `Hasher` classes and traits to a separate file, WDYT?

Reply:
I already did it in PO-16 :) Please have a look at [this commit](https://github.com/atixlabs/praos_test/commit/32ab89754b56327a0b3a467781b88920db0c80f4).